lib/src/conflicts.rs

```diff
@@ -451,24 +452,39 @@ pub async fn update_from_content(
     path: &RepoPath,
     content: &[u8],
 ) -> BackendResult<Merge<Option<FileId>>> {
+    let simplified_file_ids = file_ids.clone().simplify();
+    let simplified_file_ids = &simplified_file_ids;
+
     // First check if the new content is unchanged compared to the old content. If
     // it is, we don't need parse the content or write any new objects to the
     // store. This is also a way of making sure that unchanged tree/file
     // conflicts (for example) are not converted to regular files in the working
     // copy.
     let mut old_content = Vec::with_capacity(content.len());
-    let merge_hunk = extract_as_single_hunk(file_ids, store, path).await;
+    let merge_hunk = extract_as_single_hunk(simplified_file_ids, store, path).await;
     materialize_merge_result(&merge_hunk, &mut old_content).unwrap();
     if content == old_content {
         return Ok(file_ids.clone());
     }
 
-    let Some(hunks) = parse_conflict(content, file_ids.num_sides()) else {
-        // Either there are no self markers of they don't have the expected arity
+    // Parse conflicts from the new content, first with the arity of the simplified
+    // conflicts, then with the arity of the unsimplified conflicts if
+    // unsuccessful.
```

Comment:
I think we're trying with the unsimplified conflict only because such a conflict may already be in the working copy if it was written by an earlier version of jj. Could you say that in the comment?